engine/src/main/java/net/jqwik/engine/support/types/TypeUsageImpl.java

```diff
@@ -681,18 +641,83 @@ public Optional<TypeUsage> getSuperclass() {
 		if (rawType.getSuperclass() == null) {
 			return Optional.empty();
 		}
-		return Optional.of(TypeUsage.forType(rawType.getGenericSuperclass()));
+		// Double checking for thread safety
+		if (superclass == null) {
+			synchronized (this) {
+				if (superclass == null) {
+					superclass = createSuperclass();
+				}
+			}
+			superclass = createSuperclass();
```

Comment:
Double-creating the superclass looks weird

Reply:
it is. copy-paster error.